src/Core/Configurations/RuntimeConfigValidator.cs

```diff
@@ -830,11 +830,129 @@ public void ValidateRelationshipsInConfig(RuntimeConfig runtimeConfig, IMetadata
+                    HandleOrRecordException(new DataApiBuilderException(
+                        message: $"Entity: {entityName} has a relationship: {relationshipName} with linking target fields that are not null, " +
+                            $"but linking source fields that are null.",
+                        statusCode: HttpStatusCode.ServiceUnavailable,
+                        subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError));
+                }
+
+                if (relationship.LinkingSourceFields is not null && relationship.LinkingTargetFields is not null)
+                {
+                    if (relationship.LinkingSourceFields.Length != relationship.LinkingTargetFields.Length)
+                    {
+                        // Validation to ensure that if linking source and linking target fields exist, that they have the same number of fields. 
+                        HandleOrRecordException(new DataApiBuilderException(
+                            message: $"Entity: {entityName} has {relationship.LinkingSourceFields.Length} linking source fields defined, " +
+                                $"but {relationship.LinkingTargetFields.Length} linking target fields defined.",
+                            statusCode: HttpStatusCode.ServiceUnavailable,
+                            subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError));
+                    }
+
+                    foreach (string linkingSourceField in relationship.LinkingSourceFields)
+                    {
+                        if (!sqlMetadataProvider.TryGetBackingColumn(entityName, linkingSourceField, out _))
+                        {
+                            // Validation to ensure that entities have valid columns matching their linking source fields.
+                            HandleOrRecordException(new DataApiBuilderException(
+                                message: $"Entity: {entityName} has a relationship: {relationshipName} with linking source field: {linkingSourceField} that " +
+                                    $"does not exist as a column in {entityName}.",
+                                statusCode: HttpStatusCode.ServiceUnavailable,
+                                subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError));
+                        }
+                    }
+
+                    foreach (string linkingTargetField in relationship.LinkingTargetFields)
+                    {
+                        if (!sqlMetadataProvider.TryGetBackingColumn(relationship.TargetEntity, linkingTargetField, out _))
+                        {
+                            // Validation to ensure that entities that have linking target fields defined, define linking target fields
+                            // that are valid columns in the target entity.
```

Comment:
```suggestion
                            // Validation to ensure that linking target fields defined for a target entity, are valid columns present in the target entity.
```

Reply:
I added some more clarification. But I want to keep the language clear that the `linking target fields` are being defined in the relationship of the current entity we are looking at. We don't look at the linking target fields as defined in the relationship section of the target entity in the config, but rather the linking target fields that are defined in the current entity we are validating, and then verify they exist as backing columns for the source of the target entity.

Just want to be as clear as possible here because the usage of "target" exists in both the source and target entity, so it can be easy to confuse the two (for me at least).